### ISSUE TITLE
Feature Request:  Fallback options for remote embedding services

### ISSUE THREAD

**Issue by joel-odlund** [typesense/typesense#1757]
## Description

When using an external embedding service such as OpenAI,  Typesense becomes sensitive to outages in the external service.

## Steps to reproduce

Configure Typesense to use OpenAI.   Disable the OpenAI token.  

## Desired Behavior

It would be nice if Typesense could provide some fallback logic,   for example 
* using a secondary embedding provider
* ignoring the embedding field if needed and use the other provided fields without failiure
*
## Actual Behavior

Typesense returns an HTTP error when OpenAI is not available.

**Comment by Ku3mi41** [comment]
If you set up embedding, then you expect that it will always be filled.
- if you simply skip this stage, the embedding will never be filled in at all
- if you use a different model, there will be embedding from a different model that does not match the one used for requests

**Comment by joel-odlund** [comment]
> If you set up embedding, then you expect that it will always be filled.
> 
> * if you simply skip this stage, the embedding will never be filled in at all
Yes, if the embedding is unavailable at search time,  it could be an option to fall back to using the other fields in the search, i.e. keyword search fields. 

> * if you use a different model, there will be embedding from a different model that does not match the one used for requests

What i had in mind was to have fallback embeddings indexed along with the main embeddings, if you want to have a fallback available.  It is already possible to have multiple embedding fields and models at the same time, I believe.
